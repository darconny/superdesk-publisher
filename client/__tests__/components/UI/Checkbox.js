import React from 'react'
import Checkbox from '../../../components/UI/Checkbox'
import { render, fireEvent } from '@testing-library/react'

describe('UI/Checkbox', () => {
    it('renders correctly', () => {
    const {container} = render(<Checkbox/>)

    expect(container.firstChild).toMatchSnapshot()
    })

    it('onChange function fired', () => {
        const onChange = jest.fn()
        const { container } = render(<Checkbox onChange={onChange}/>)
        const button = container.querySelector('.sd-checkbox')

        fireEvent.click(button)
        expect(onChange).toHaveBeenCalled()
    })
})