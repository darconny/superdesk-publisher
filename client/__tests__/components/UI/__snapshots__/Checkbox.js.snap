// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI/Checkbox renders correctly 1`] = `
<span
  class="sd-check__wrapper"
>
  <span
    class="sd-checkbox"
  />
  <label
    class=""
    style="color: rgb(255, 255, 255);"
  />
</span>
`;
